app/questionnaire_state/state_repeating_answer_question.py

```diff
@@ -44,6 +47,25 @@ def create_new_answer_state(self, answer_schema, answer_instance, group_instance
     def add_new_answer_state(self, answer_state):
         self.answers.append(answer_state)
 
+    def answers_grouped_by_instance(self):
+        """
+        Groups answers by their answer_instance Id.
+        :return: A list of lists containing the answers grouped by answer_instance.
+        """
+
+        answers_by_id = defaultdict(list)
+        for answer in self.answers:
+            answers_by_id[answer.id].append(answer)
+
+        answers_by_answer_instance = OrderedDict()
```

Comment:
It would read more clearly if this variable echoed the method name i.e. `answer_states_grouped_by_instance`